### PR TITLE
Add DeviceInfo API

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change ###

Added the DeviceInfo APIs from #9.

### Bugs Fixed ###

 - No bugs here :)

### API Changes ###

Added: 
 
 - See #9 

### Behavioral Changes ###

 - New features.

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Caboodle/DeviceInfo/DeviceInfo.android.cs [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+            return appVersionString;
+        }
+
+        static string GetAppBuild()
+        {
+            SetAppVersions();
+            return appBuild;
+        }
+
+        static string GetPlatform() => Platforms.Android;
```
Comment (jamesmontemagno) [inline Caboodle/DeviceInfo/DeviceInfo.android.cs]: Why is this a method? Can't it just be Platform?

File: Caboodle/DeviceInfo/DeviceInfo.netstandard.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+﻿namespace Microsoft.Caboodle
+{
+    public static partial class DeviceInfo
```
Comment (jamesmontemagno) [inline Caboodle/DeviceInfo/DeviceInfo.netstandard.cs]: These should be properties, not methods.

File: Caboodle/DeviceInfo/DeviceInfo.android.cs (hunk 1 above)
Comment (mattleibow) [reply Caboodle/DeviceInfo/DeviceInfo.android.cs]: It is a property, in the shared file. This is just the private implementation.

File: Caboodle/DeviceInfo/DeviceInfo.netstandard.cs (hunk 2 above)
Comment (mattleibow) [reply Caboodle/DeviceInfo/DeviceInfo.netstandard.cs]: private implementations.

Comment (jamesmontemagno) [reply Caboodle/DeviceInfo/DeviceInfo.netstandard.cs]: AHHHHHHHHHHHHHHHHHHHHHHHHHHHHH

File: Caboodle/DeviceInfo/DeviceInfo.shared.cs [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        }
+
+        public static class Platforms
+        {
+            // try to match Xamarin.Forms:
+            // https://github.com/xamarin/Xamarin.Forms/blob/2.5.1/Xamarin.Forms.Core/Device.cs#L14-L19
+
+            public const string iOS = "iOS";
+            public const string Android = "Android";
+            public const string UWP = "UWP";
```
Comment (Redth) [inline Caboodle/DeviceInfo/DeviceInfo.shared.cs]: I think this string doesn't match forms ? Can you double check the values?

File: Caboodle/Caboodle.csproj [hunk 4]
```diff
@@ -2,8 +2,8 @@
   <PropertyGroup>
     <!--Work around so the conditions work below-->
     <TargetFrameworks></TargetFrameworks>
-    <TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">netstandard1.0;netstandard2.0;Xamarin.iOS10;MonoAndroid80;uap10.0.16299</TargetFrameworks>
-    <TargetFrameworks Condition=" '$(OS)' != 'Windows_NT' ">netstandard1.0;netstandard2.0;Xamarin.iOS10;MonoAndroid80</TargetFrameworks>
+    <TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">netstandard1.0;netstandard1.4;netstandard2.0;Xamarin.iOS10;MonoAndroid80;uap10.0.14393;uap10.0.16299</TargetFrameworks>
```
Comment (Redth) [inline Caboodle/Caboodle.csproj]: Why 1.4 now as well?

File: Caboodle/DeviceInfo/DeviceInfo.shared.cs (hunk 3 above)
Comment (mattleibow) [reply Caboodle/DeviceInfo/DeviceInfo.shared.cs]: it does. see the link above.

**Comment by mattleibow** [comment]
In this PR, i took the liberty of making the existing samples look nicer with padding and other visual separators (such as bold text)

**Comment by jamesmontemagno** [comment]
I think i broke and merged incorrect :(

**Pull request merged**